src/Raven.Server/Documents/Sharding/ShardedDatabaseContext.cs

```diff
@@ -82,6 +83,18 @@ public ShardedDatabaseContext(ServerStore serverStore, DatabaseRecord record)
         public void UpdateDatabaseRecord(RawDatabaseRecord record, long index)
         {
             UpdateConfiguration(record.Settings);
+            
+            if (AreDictionaryKeysEqual(record.Sharding.Shards, _record.Sharding.Shards) == false)
+            {
+                ShardExecutor.Dispose();
```

Comment:
this should be in `using` otherwise we will have a brief moment without any executor

Reply:
done